go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -353,62 +443,77 @@ func (npw *nodePortWatcher) updateServiceInfo(index ktypes.NamespacedName, servi
-		npw.updateServiceFlowCache(service, false, false)
+		npw.updateServiceFlowCache(service, addFlows, false)
 		npw.ofm.requestFlowSync()
 		if !npw.dpuMode {
-			// Always try and delete all rules here
-			delSharedGatewayIptRules(service, true)
-			delSharedGatewayIptRules(service, false)
+			// Always try and delete all rules here in full mode & in host only mode. We don't touch iptables in dpu mode.
+			// +--------------------------+-----------------------+--------------+--------------------------------+
+			// | svcHasLocalHostNetEndPnt | ExternalTrafficPolicy | GatewayMode  |     Scenario for deletion      |
+			// |--------------------------|-----------------------|--------------|--------------------------------|
+			// |                          |                       |              |      deletes the REDIRECT      |
+			// |         true             |          local        | shared/local |      rules for etp=local +     |
+			// |                          |                       |              |      host-networked eps        |
+			// |--------------------------|-----------------------|--------------|--------------------------------|
+			// |                          |                       |              | deletes the DNAT rules for     |
+			// |         false            |          local        |   local      | etp=local + non-local-host-net |
+			// |                          |                       |              | eps towards masqueradeIP       |
+			// |--------------------------|-----------------------|--------------|--------------------------------|
+			// |                          |                       |              |                                |
+			// |         false            |          cluster      | shared/local |   	deletes the DNAT rules    |
+			// |         false            |           local       |   shared     |         towards clusterIP      |
+			// |                          |                       |              |       for the default case     |
+			// +--------------------------+-----------------------+--------------+--------------------------------+
+
+			// case1: deletes the REDIRECT rules for etp=local + host-networked pods in both gwmodes
+			delGatewayIptRules(service, true)
+			// case2Orcase3a/3b:
+			// deletes the DNAT rules towards masqueradeIP for etp=local + ovn-k pods in LGW mode OR
+			// deletes the DNAT rules towards clusterIP for etp=local + ovn-k pods in SGW mode OR
+			// deletes the DNAT rules towards clusterIP for etp=cluster in both gwmodes
+			delGatewayIptRules(service, false)
 		}
 		return
 	}
 
-	// For host only node always try and delete rules here
-	// externalTrafficPolicy is not implemented for dpu mode
-	delSharedGatewayIptRules(service, false)
+	// For host only mode always try and delete all rules here
+	// case1: deletes the REDIRECT rules for etp=local + host-networked pods in both gwmodes
+	delGatewayIptRules(service, true)
+	// case2Orcase3a/3b:
```

Comment:
typo?

Reply:
oops not sure where all the spaces flew off too...